### PR TITLE
Handle string list for extra siblings of description

### PR DESCRIPTION
**Pull request by itzg**
As observed in https://github.com/itzg/docker-minecraft-server/issues/2545 the `description` field of a ping response can sometimes have an `extra` siblings which is a simple list of strings, such as this response from Paper 1.20.4:

```json
{"version":{"name":"Paper 1.20.4","protocol":765},"enforcesSecureChat":true,"description":{"text":"Foo","extra":["Bar"]},"players":{"max":20,"online":0}}
```

This behavior seems to be specifically tracking changes announced by Mojang in https://www.minecraft.net/en-us/article/minecraft-java-edition-1-20-3.

I have tested this change for backward compatibility where Paper 1.20.2 responds with

```json
{"version":{"name":"Paper 1.20.2","protocol":764},"enforcesSecureChat":true,"description":{"extra":[{"text":"Bar"}],"text":"Foo"},"players":{"max":20,"online":0}}
```

### CODE REVIEW

**Comment by raqbit** [comment]
Nice catch! Can you add a testcase for this scenario?

**Comment by itzg** [comment]
Certainly. Sorry, I hadn't noticed the tests file up a level.

File: .github/workflows/test.yml
Comment (itzg) [inline .github/workflows/test.yml:1]: I was looking for my test changes to pass...and noticed there wasn't a workflow yet 😀

File: status_test.go [hunk 1]
```diff
@@ -1,77 +1,112 @@
...
+		VersionName       string
+		Protocol          int32
+		Description       string
+		MaxPlayers        int32
+		OnlinePlayers     int32
+		PlayerSampleCount int
+		PlayerSampleName  string
+		PlayerSampleUuid  string
+		Favicon           string
+	}{
```
Comment (itzg) [inline status_test.go:25]: I hope you don't mind, but I thought I'd change this to a table driven test. If you would prefer I can revert this and add the new scenario as a separate test case.

File: .github/workflows/test.yml
Comment (raqbit) [reply .github/workflows/test.yml:1]: Yup, I only realized it when you sent this PR!

**Comment by itzg** [comment]
I just wanted to see if you had any concerns about the test related changes.

**Review by raqbit — COMMENTED** [review]
Can you add newlines at the end of the JSON file & actions YAML?

Otherwise it looks good to me. Thanks for adding CI as well!

**Comment by itzg** [comment]
> Can you add newlines at the end of the JSON file & actions YAML?

Certainly! I really should setup IntelliJ to enforce that for me 😄

**Comment by raqbit** [comment]
~~Oh, it appears the os.ReadFile API was only added in Go 1.16. Can you revert that change in a follow-up? I did not notice that the CI did not run as part of this MR 😬~~

I see the go.mod file specifies Go 1.12, which we probably shouldn't support anymore. I'll update it & add matrix testing to the CI for Go 1.16+.

**Comment by itzg** [comment]
Oh, I'm sorry about that. I guess the CI didn't run for this PR since it was newly introduced in the branch. 

I'd actually recommend keeping the go.mod compatibility at 1.12 and I revert the change I made since then it advertises the backward compatibility of the library.

**Comment by itzg** [comment]
...but using a modern version of Go is good too 😀

**Pull request merged**